src/util/global_worker_pool.js

```diff
@@ -15,3 +15,24 @@ export default function getGlobalWorkerPool () {
     }
     return globalWorkerPool;
 }
+
+export function preloadWorkerPool() {
+    const workerPool = getGlobalWorkerPool();
+    workerPool.acquire(PRELOAD_POOL_ID);
+}
+
+export function removePreloadedWorkerPool() {
+    const pool = globalWorkerPool;
+    if (!pool) {
+        console.warn('Attempted to remove preloaded worker pool before creating one, skipping. Use mapboxggl.preloadWorkerPool() to preload a global worker pool');
+        return;
+    }
+
+    // Remove the pool only if all maps that referenced the preloaded global worker pool have been removed.
+    if (pool.isPreloaded() && pool.numActive() === 1) {
```

Comment:
Instead of relying on the user to remove worker pools, we could flag the pool for deletion (refcount style) and delete it once the last map instance disappears. It looks like the workers are automatically terminated if we release `PRELOAD_POOL_ID`.

Reply:
We do that already, but that means that a developer wont be able to cache workers between map tear down and re-initialize, which is where most of the performance gain is.
Eg. In studio go from style list -> a style  -> back to style list -> to a different style.